src/client/components/builder/ConstantsTab.tsx

```diff
@@ -0,0 +1,158 @@
+  id: `T${id}`,
+  title: 'Insert map title',
+  table: [{ key: '', value: NaN }],
+})
+
+export const ConstantsTab: FC = () => {
+  const [activeIndex, setActiveIndex] = useState<number>(0)
+  const [offsetTop, setOffsetTop] = useState<number>(16)
+  const [nextUniqueId, setNextUniqueId] = useState<number>(1)
+  const { config, dispatch } = useCheckerContext()
+
+  const { constants } = config
+
+  useEffect(() => {
+    let highestIndex = 0
+    config.constants.forEach((constant) => {
+      const constantIndex = parseInt(constant.id.slice(1))
+      highestIndex = Math.max(highestIndex, constantIndex)
+    })
+    setNextUniqueId(highestIndex + 1)
+  }, [config])
+
+  const toolbarOptions = [
+    {
+      icon: <BiPlusCircle />,
+      label: 'Add map',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Add,
+          payload: {
+            element: generateDefaultMap(nextUniqueId),
+            configArrName: ConfigArrayEnum.Constants,
+            newIndex: activeIndex + 1,
+          },
+        })
+        setActiveIndex(activeIndex + 1)
+        setNextUniqueId(nextUniqueId + 1)
+      },
+    },
+    {
+      icon: <BiUpArrowAlt />,
+      label: 'Move up',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Reorder,
+          payload: {
+            currIndex: activeIndex,
+            newIndex: activeIndex - 1,
+            configArrName: ConfigArrayEnum.Constants,
+          },
+        })
+        setActiveIndex(activeIndex - 1)
+      },
+      disabled: activeIndex === 0,
+    },
+    {
+      icon: <BiDownArrowAlt />,
+      label: 'Move down',
+      onClick: () => {
+        dispatch({
+          type: BuilderActionEnum.Reorder,
+          payload: {
+            currIndex: activeIndex,
+            newIndex: activeIndex + 1,
+            configArrName: ConfigArrayEnum.Constants,
+          },
+        })
+        setActiveIndex(activeIndex + 1)
+      },
+      disabled: activeIndex === constants.length - 1,
+    },
+  ]
+
+  const onSelect = ({ index }: { index: number }) => {
+    setActiveIndex(index)
+  }
+
+  const onActive = ({ top }: { top: number }) => {
+    setOffsetTop(top)
+  }
+
+  const addInitialMap = () => {
+    dispatch({
+      type: BuilderActionEnum.Add,
+      payload: {
+        element: generateDefaultMap(0),
+        configArrName: ConfigArrayEnum.Constants,
+        newIndex: 1,
+      },
+    })
+    setActiveIndex(0)
+    setNextUniqueId(1)
+  }
+
+  const renderMap = (constant: checker.Constant, index: number) => {
+    const commonProps = {
+      key: constant.id + index,
+      id: constant.id,
+      active: activeIndex === index,
+      data: constant,
+      index,
+      onActive,
+      onSelect,
+      setActiveIndex,
+      nextUniqueId,
+      setNextUniqueId,
+    }
+    return <MapTable {...commonProps} />
+  }
+
+  return (
+    <VStack align="stretch" position="relative" spacing={4}>
```

Comment:
```suggestion
    <>
        <Alert status="info">Constants can only be numeric values...</Alert>
        <VStack align="stretch" position="relative" spacing={4}>
```

We should add the info block to give some context to the constants tab.

![image](https://user-images.githubusercontent.com/3666479/109249592-f637b800-7822-11eb-8aef-efd14d39bc19.png)

Reply:
addressed with e243cb3

Reply:
<img width="937" alt="Screenshot 2021-02-26 at 6 08 35 PM" src="https://user-images.githubusercontent.com/19917616/109286771-aa563480-785d-11eb-8c88-4fdeda73ee18.png">